Assets/Scripts/Tank/TankBotStates/AttackingState.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace TankBattle.Tank
+{
+    public class AttackingState : TankBotState
+    {
+        private TankController targetTank;
+        public TankController TargetTank {set { targetTank = value; }}
+        public override void OnEnterState(TankController _tankController, TankView _tankView)
+        {
+            base.OnEnterState(_tankController, _tankView);
+        }
+
+        public override void OnExitState()
+        {
+            base.OnExitState();
+        }
+
+        public void Update() 
+        {
+            if (targetTank != null && (Vector3.Distance(targetTank.GetTankPosition(), tankController.GetTankPosition()) < TankService.Instance.BotTankPropScriptableObject.AttackTriggerDistance))
```

Comment:
you can simplify this if condition 
if( TT != null && IsTargetTankInFiringRange(TT)) {
// makes it more readable and easy to understand 
}